src/main/java/com/direwolf20/buildinggadgets/common/tools/ExchangingModes.java

```diff
@@ -16,26 +15,154 @@
+        } else {
+            for (BlockPos pos : BlockPos.getAllInBox(area.minX, area.minY, area.minZ, area.maxX, area.maxY, area.maxZ)) {
+                if (isReplaceable(world, pos, current, target, fuzzyMode)) {
+                    coordinates.add(pos);
+                }
+            }
+        }
+
+        return coordinates;
+    }),
+    VerticalColumn((world, player, hit, sideHit, tool) -> {
+        List<BlockPos> coordinates = new ArrayList<>();
+        IBlockState target = GadgetUtils.getToolActualBlock(tool);
+        IBlockState current = world.getBlockState(hit);
+        int range = GadgetUtils.getToolRange(tool);
+        int radius = (range - 1) / 2;
+        boolean fuzzyMode = GadgetGeneric.getFuzzy(tool);
+
+        BlockPos base;
+        EnumFacing extension;
+        if (sideHit.getAxis().isVertical()) {
+            EnumFacing playerFacing = player.getHorizontalFacing();
+            base = hit.offset(playerFacing.getOpposite(), radius);
+            extension = playerFacing;
+        } else {
+            base = hit.down(radius);
+            extension = EnumFacing.UP;
+        }
+
+        for (int i = 0; i < range; i++) {
+            BlockPos pos = base.offset(extension, i);
+            if (isReplaceable(world, pos, current, target, fuzzyMode)) {
+                coordinates.add(pos);
+            }
+        }
+
+        return coordinates;
+    }),
+    HorizontalColumn((world, player, hit, sideHit, tool) -> {
+        List<BlockPos> coordinates = new ArrayList<>();
+        IBlockState target = GadgetUtils.getToolActualBlock(tool);
+        IBlockState current = world.getBlockState(hit);
+        int range = GadgetUtils.getToolRange(tool);
+        int radius = (range - 1) / 2;
+        boolean fuzzyMode = GadgetGeneric.getFuzzy(tool);
+
+        EnumFacing playerFacing = player.getHorizontalFacing();
+        EnumFacing extension = playerFacing.rotateY();
+        BlockPos base = hit.offset(extension.getOpposite(), radius);
+
+        for (int i = 0; i < range; i++) {
+            BlockPos pos = base.offset(extension, i);
+            if (isReplaceable(world, pos, current, target, fuzzyMode)) {
+                coordinates.add(pos);
+            }
+        }
+
+        return coordinates;
+    }),
+    Grid((world, player, hit, sideHit, tool) -> {
+        List<BlockPos> coordinates = new ArrayList<>();
+        IBlockState target = GadgetUtils.getToolActualBlock(tool);
+        IBlockState current = world.getBlockState(hit);
+        int range = GadgetUtils.getToolRange(tool) + 1;
+        boolean fuzzyMode = GadgetGeneric.getFuzzy(tool);
+
+        for (int x = range * -7 / 5; x <= range * 7 / 5; x++) {
+            for (int z = range * -7 / 5; z <= range * 7 / 5; z++) {
+                if (x % (((range - 2) % 6) + 2) == 0 && z % (((range - 2) % 6) + 2) == 0) {
+                    BlockPos pos = new BlockPos(hit.getX() + x, hit.getY(), hit.getZ() + z);
```

Comment:
Whilst you're at it: these BlockPos everywhere should be MutableBlockPos for performance reasons...

Reply:
Then we need to re-implement methods offset, which returns a immutable BlockPos. I think I'll make a new MutableBlockPos implementation since the vanilla one is from some ancient version

Reply:
Also they technically can't be mutable since we need all of them in the coordinates list. What I could do is calculate the coordinates using mutable, and then make a immutable copy for every valid position.

Reply:
You're correct... Though I don't like these coordinate lists... I might wind up changing that to calling a consumer on the correct position instead of parsing a list of coordinates and then parse it again to do something with the coordinates.

Doing that is beyond the scope of this PR though